packages/neos-ui/src/Containers/RightSideBar/Inspector/index.js

```diff
@@ -96,14 +104,28 @@ export default class Inspector extends PureComponent {
     }
 
     render() {
-        const {focusedNode, nodeTypesRegistry, transientValues} = this.props;
+        const {focusedNode, nodeTypesRegistry, transientValues, node, validatorRegistry, commit} = this.props;
 
         if (!focusedNode) {
             return this.renderFallback();
         }
 
+        const nodeType = nodeTypesRegistry.get(node.nodeType);
+        if (!nodeType.properties) {
+            console.error(`No properties configured in ${node.nodeType} nodetype`);
+        }
+        let validationErrors = null;
+        if (transientValues) {
+            const transientProps = {};
+            Object.keys(transientValues.toJS()).forEach(key => {
+                transientProps[key] = $get([key, 'value'], transientValues);
+            });
+            validationErrors = validate(transientProps, nodeType.properties, validatorRegistry);
+        }
+
         const viewConfiguration = nodeTypesRegistry.getInspectorViewConfigurationFor(focusedNode.nodeType);
-        const isDisabled = transientValues === undefined;
+        const isDiscardDisabled = transientValues === undefined;
+        const isApplyDisabled = transientValues === undefined || validationErrors !== null;
```

Comment:
Not sure it's a good idea. That would be D.R.Y. for wrong reasons. The logic for `isDiscardDisabled` may change anytime, there's no semantical dependency between the states of these two buttons.